### PR TITLE
[BUGFIX] Respect page language overlay

### PR DESCRIPTION
**Pull request by froehlit**
In `\FluidTYPO3\Vhs\ViewHelpers\Resource\Record\FalViewHelper::getFileReferences`  use `$record['_PAGES_OVERLAY_UID']` to get the correct page uid for the language overlay.

This solution works for me, but I'm not an expert in TYPO3s language handling.

### CODE REVIEW

File: Classes/ViewHelpers/Resource/Record/FalViewHelper.php [hunk 1]
```diff
@@ -118,6 +118,8 @@ protected function getFileReferences($table, $field, $uidOrRecord)
             $sqlRecordUid = $record['t3ver_oid'];
         } elseif (isset($record['_LOCALIZED_UID'])) {
             $sqlRecordUid = $record['_LOCALIZED_UID'];
+        } elseif (isset($record['_PAGES_OVERLAY_UID'])) {
```
Comment (NamelessCoder) [inline Classes/ViewHelpers/Resource/Record/FalViewHelper.php:121]: This property name is specific to pages and does not belong in the generic FAL resource ViewHelper. A more appropriate place is in the page-specific FAL resource ViewHelper.

Comment (froehlit) [reply Classes/ViewHelpers/Resource/Record/FalViewHelper.php:121]: If i fixe it only in `v:page.resource.fal` viewhelper, the problem remains, if you call the `v:resource.record.fal` viewhelper with `table="page"` as an argument.

File: Classes/ViewHelpers/Page/Resources/FalViewHelper.php [hunk 2]
```diff
@@ -36,6 +36,11 @@ class FalViewHelper extends ResourcesFalViewHelper
      */
     protected $field = self::DEFAULT_FIELD;
 
+    /**
+     * @var string
+     */
+    protected $localizedIdField = "_PAGES_OVERLAY_UID";
```
Comment (NamelessCoder) [inline Classes/ViewHelpers/Page/Resources/FalViewHelper.php:42]: I'm not sure this works the way you think. On TYPO3v8 this should be the UID of the `pages_language_overlay` record (correct me if I'm wrong!) - and...

File: Classes/ViewHelpers/Resource/Record/FalViewHelper.php [hunk 3]
```diff
@@ -61,6 +61,11 @@ class FalViewHelper extends AbstractRecordResourceViewHelper
      */
     protected $escapeOutput = false;
 
+    /**
+     * @var string
+     */
+    protected $localizedIdField = "_LOCALIZED_UID";
```
Comment (NamelessCoder) [inline Classes/ViewHelpers/Resource/Record/FalViewHelper.php:67]: ...on all TYPO3 versions this field should contain the UID of the original record that was translated, not the translated record's UID (which should be in the `uid` property). For non-page records this part should work, assuming that `uid` contains the `uid` of the non-default language record (the methods called with this variable require the original record's UID if I remember it right), but for pages I think the implementation with `_PAGES_OVERLAY_UID` breaks on TYPO3v8 as it doesn't contain the UID of a page, but rather the UID of a `pages_language_overlay` record.

**Comment by user123801** [comment]
@NamelessCoder Any update on the issue? The problem persists in the latest version of the vhs extension, it shows the images of the default language in alternative site languages.

**Comment by fnagel** [comment]
Update: still having issues when using slide and having disabled pages on the rootline.

**Comment by NamelessCoder** [comment]
Fixed by https://github.com/FluidTYPO3/vhs/commit/169a6112fb4625a86c5bcb3b05a7c3b7df361d77

**Pull request closed without merging**